Pipfile

```diff
@@ -4,6 +4,9 @@ verify_ssl = true
 name = "pypi"
 
 [packages]
+isort = "*"
+black = "*"
+pre-commit = "*"
```

Comment:
Is there a way to automatically lock the packages? I cannot seem to find any. Or do you have to do it one by one manually?

Reply:
there's no way to auto lock the packages directly in the pip file. There's a bit of a work around involving the use of [pip tools](https://pypi.org/project/pip-tools/) to generate a `requirements.txt` file with the latest versions set explicitly, you can then use this `requirements.txt` file to update the pipfile, then delete the `requirements.txt` file. 

In most cases it's just easier to google the latest version of the packages yourself and then specifiy it manually. 

In any case, I should mention that the pipfile itself isn't the file pipenv uses to create the virtual environment, the pipfile.lock is. The pipfile.lock file actually _does_ have explicit version locking (hence the name pipfile.lock) and is generated automatically when you run `pipenv install`. The pipfile is just there to be human readable, so you technically don't need to use it to create the virtual environment as long as you have the pipenv.lock file. However it's best practice to use explicitly declare versions in the pipfile itself, this is because in some cases like in the case of a mono-repo you might have a script that generates the pipfile.lock for other developers instead of sharing the pipfile.lock with them. Another reason to have them explicitly mentioned in the pipfile is because it allows developers to more easily understand which version of a package they’re dealing with during development.

That being said, after updating the pipfile to have explicit version names mentioned, make sure to run `pipenv lock` to make sure the pipenv.lock file is updated as well.

Reply:
Okay, makes sense. Thanks for the detailed explanation.